File: medium/array/Convert an Array Into a 2D Array With Conditions.cpp
class Solution {
public:
    vector<vector<int>> findMatrix(vector<int>& nums) {
        vector<vector<int>> ans;
        unordered_set<int> s;
        while(!nums.empty()){
            vector<int> row;
            for(int i=0;i<nums.size();i++){
                if(s.find(nums[i]) == s.end()) {
                    s.insert(nums[i]);
                    row.push_back(nums[i]);
                    nums.erase(nums.begin()+i);
                    i--;
                }
            }ans.push_back(row);
            s.clear();
        }return ans;
    }
};

File: medium/array/Convert an Array Into a 2D Array With Conditions.py
class Solution(object):
    def findMatrix(self, nums):
        ans = list()
        s = set()
        while nums:
            row = list()
            i=0
            while i<len(nums):
                if nums[i] not in s:
                    s.add(nums[i])
                    row.append(nums[i])
                    nums.pop(i)
                    i-=1
                i+=1
            ans.append(row)
            s.clear()
        return ans



